File: benchmark/delphi/task5_class.pas
unit PointUnit;

interface

type
  TPoint = class
  private
    FX, FY: Double;
  public
    constructor Create(X, Y: Double);
    function DistanceTo(Other: TPoint): Double;
    property X: Double read FX;
    property Y: Double read FY;
  end;

implementation

uses Math;

constructor TPoint.Create(X, Y: Double);
begin
  FX := X;
  FY := Y;
end;

function TPoint.DistanceTo(Other: TPoint): Double;
var
  DX, DY: Double;
begin
  DX := FX - Other.X;
  DY := FY - Other.Y;
  Result := Sqrt(DX * DX + DY * DY);
end;

end.

File: benchmark/visualbasic/task5_class.vb
Public Class Point
    Public Property X As Double
    Public Property Y As Double

    Public Sub New(x As Double, y As Double)
        Me.X = x
        Me.Y = y
    End Sub

    Public Function DistanceTo(other As Point) As Double
        Dim dx As Double = X - other.X
        Dim dy As Double = Y - other.Y
        Return Math.Sqrt(dx * dx + dy * dy)
    End Function
End Class
